src/main/java/Storage.java

```diff
@@ -0,0 +1,82 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+class Storage {
+    /**
+     * handles then loading of hard-disk data and saving of the data when task list is updated.
+     */
+
+    private String filepath;
+    private ArrayList<Task> taskList;
+
+    /**
+     * constructor, takes in path of the file.
+     * @param filepath of the file.
+     */
+    Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * loads the data from hard-disk file.
+     * @return an array list of task to be passed into TaskList.
+     * @throws FileNotFoundException when input format is different from required.
+     * @throws ParseException when input format is different from required.
+     */
+    ArrayList<Task> load() throws FileNotFoundException, ParseException {
+
+        ArrayList<Task> taskList = new ArrayList<>();
+        File f = new File(filepath); // create a File for the given file path
+        Scanner sc = new Scanner(f); // create a ScanneZZ:1q!r using the File as the source
+        while (sc.hasNext()) {
+            Task t = new Task(""); // dummie task
```

Comment:
Dummy task not really needed. Can just throw an error in the default switch block if there is no type.